src/App.tsx

```diff
@@ -5,7 +5,21 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const handleMovieAdd = (newMovie: Movie) => {
+    const isMovieAdded = movies.find(
+      (movie) => movie.imdbId === newMovie.imdbId,
+    );
+
+    if (isMovieAdded) {
+      return false;
+    }
+
+    setMovies((current) => [...current, newMovie]);
+
+    return true;
+  };
```

Comment:
Why you return value if you dont use it?
```suggestion
  const handleMovieAdd = (newMovie: Movie) => {
    const isMovieAdded = movies.find(
      (movie) => movie.imdbId === newMovie.imdbId,
    );

    if (!isMovieAdded) {
     setMovies((current) => [...current, newMovie]);
    }
  };
```